x/staking/types/validator.go

```diff
@@ -421,7 +422,18 @@ func (v Validator) GetMoniker() string          { return v.Description.Moniker }
 func (v Validator) GetStatus() sdk.BondStatus   { return v.Status }
 func (v Validator) GetOperator() sdk.ValAddress { return v.OperatorAddress }
 func (v Validator) GetConsPubKey() crypto.PubKey {
-	return sdk.MustGetPubKeyFromBech32(sdk.Bech32PubKeyTypeConsPub, v.ConsensusPubkey)
+	// The way things are refactored now, v.ConsensusPubkey is sometimes a TM
+	// ed25519 pubkey, sometimes our own ed25519 pubkey. This is very ugly and
+	// inconsistent.
```

Comment:
could you explain when it can be a `sdk ed25519`?. Aren't we always using `sdk ed25519` due to serialization?

Reply:
I think @amaurymartiny just meant that we are primarily using `sdk ed25519` except where we explicitily need `tendermint ed25519`, when calling `encoding.PubKeyToProto` for instance.

Reply:
Then let's update the comment please.